test/jdk/java/awt/print/PrinterJob/PrinterDevice.java

```diff
@@ -54,39 +66,52 @@ public static void main(String args[]) throws PrinterException {
+        }
     }
 
     public int print(Graphics g, PageFormat pf, int pageIndex) {
-         if (pageIndex > 0 ) {
-             return Printable.NO_SUCH_PAGE;
-         }
-
-         /* Make sure calls to get DeviceConfig, its transforms,
-          * etc all work without exceptions and as expected */
-         Graphics2D g2 = (Graphics2D)g;
-         GraphicsConfiguration gConfig = g2.getDeviceConfiguration();
-         AffineTransform dt = gConfig.getDefaultTransform();
-         AffineTransform nt = gConfig.getNormalizingTransform();
-         AffineTransform gt = g2.getTransform();
+        if (pageIndex > 0) {
+            return Printable.NO_SUCH_PAGE;
+        }
 
-         System.out.println("Graphics2D transform = " + gt);
-         System.out.println("Default transform = " + dt);
-         System.out.println("Normalizing transform = " + nt);
+        Graphics2D g2 = (Graphics2D)g;
+        GraphicsConfiguration gConfig = g2.getDeviceConfiguration();
+        AffineTransform nt = null;
+        try {
+            /* Make sure calls to get DeviceConfig, its transforms,
+             * etc all work without exceptions and as expected */
+            System.out.println("GraphicsConfig="+gConfig);
+            AffineTransform dt = gConfig.getDefaultTransform();
+            System.out.println("Default transform = " + dt);
+            nt = gConfig.getNormalizingTransform();
+            System.out.println("Normalizing transform = " + nt);
+            AffineTransform gt = g2.getTransform();
+            System.out.println("Graphics2D transform = " + gt);
+        } catch (Exception e) {
+            failed = true;
+            System.err.println("Unexpected exception getting transform.");
+            e.printStackTrace();
+            throw e;
+        }
 
-         Rectangle bounds = gConfig.getBounds();
-         System.out.println("Bounds = " + bounds);
-         if (!nt.isIdentity()) {
-             throw new RuntimeException("Expected Identity transdform");
-         }
+        Rectangle bounds = gConfig.getBounds();
+        System.out.println("Bounds = " + bounds);
+        if (!nt.isIdentity()) {
+            failed = true;
+            throw new RuntimeException("Expected Identity transform");
+        }
 
-         /* Make sure that device really is TYPE_PRINTER */
-         GraphicsDevice gd = gConfig.getDevice();
-         System.out.println("Printer Device ID = " + gd.getIDstring());
-         if (!(gd.getType() == GraphicsDevice.TYPE_PRINTER)) {
-             throw new RuntimeException("Expected printer device");
-         }
-         System.out.println(" *** ");
-         System.out.println("");
-         return Printable.PAGE_EXISTS;
+        /* Make sure that device really is TYPE_PRINTER */
+        GraphicsDevice gd = gConfig.getDevice();
+        System.out.println("Printer Device ID = " + gd.getIDstring());
+        if (!(gd.getType() == GraphicsDevice.TYPE_PRINTER)) {
```

Comment:
```suggestion
        if (gd.getType() != GraphicsDevice.TYPE_PRINTER) {
```
It looks cleaner this way. I missed it when I went over the code the first time.